sdks/python/apache_beam/io/mongodbio_pipeline.py

```diff
@@ -0,0 +1,85 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import argparse
+import logging
+import time
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+
+def run(argv=None):
+  default_db = 'beam_mongodbio_it_db'
+  default_coll = 'integration_test_%d' % time.time()
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--mongo_uri',
+                      default='mongodb://localhost:27017',
+                      help='mongo uri string for connection')
+  parser.add_argument('--mongo_db',
+                      default=default_db,
+                      help='mongo uri string for connection')
+  parser.add_argument('--mongo_coll',
+                      default=default_coll,
+                      help='mongo uri string for connection')
+  parser.add_argument('--num_documents',
+                      default=1000,
+                      help='The expected number of documents to be generated '
+                      'for write or read',
+                      type=int)
+  parser.add_argument('--batch_size',
+                      default=100,
+                      help=('batch size for writing to mongodb'))
+  known_args, pipeline_args = parser.parse_known_args(argv)
+
+  # Test Write to MongoDB
+  with TestPipeline(options=PipelineOptions(pipeline_args)) as p:
+    logging.info('Writing %d documents to mongodb' % known_args.num_documents)
+    docs = [{'test_document_n': x} for x in range(known_args.num_documents)]
```

Comment:
I'm confused by this statement, is x here the content of the document ? Can we add a more representative content that we can later use to query/verify ?

Reply:
make sense will do.